### DISCUSSION TITLE
blog/wolfi-os-for-php/

### DISCUSSION THREAD

**Discussion by giscus** [shyim/shyim.me#57] [Q&A]
# blog/wolfi-os-for-php/

Currently, the de-facto standard to Dockerizing PHP applications is to use the Docker PHP official image. This is maintained by Docker itself and community members and has compiled PHP with no extensions. To install extensions, we have to do this in our Dockerfile:
FROMphp:8.2-cliRUN docker-php-ext-install pdo pdo_mysql curlThe helper script docker-php-ext-install does unpack the PHP source code and installs the given extensions. For some extensions like gd, we need also libraries installed like libpng and so on.

https://shyim.me/blog/wolfi-os-for-php/

**Comment by AubreyHewes** [comment]
Love it!

Is this also the base for shopware saas solutions?

**Comment by souhaiebtar** [comment]
"For additional packages, I set up an own repository " mistake, it should be "i set up my own repository", i really like your work ,

**Comment by pounard** [comment]
I am sorry but you are comparing apples and oranges. When you add commands into a Dockerfile each one will create a new layer: if your command doesn't do its cleanup in the same command line, then the layer will add up all the temporary files used in between.

Fact is, using `debian:trixie-slim` image, I managed to end with a 140M image which contains the exact same dependencies as your Wolfi image, which is **not** *557M*.

It took me less than 20 minutes to get this result, and I'm sure it can be improved even further. At this point, I don't see any benefit in depending upon some private company, which is a real security and confidentiality issue considering I more or less get to same result in a few minutes with community images.

Here is the Dockerfile:

```
FROM debian:trixie-slim

RUN apt-get update \
    && DEBIAN_FRONTEND=noninteractive apt-get install -y --no-install-recommends -o Dpkg::Options::='--force-confdef' -o Dpkg::Options::='--force-confold' \
    php8.4-pdo \
    php8.4-pdo-mysql \
    php8.4-gd \
    php8.4-curl \
  # Remove some useless packages
  && DEBIAN_FRONTEND=noninteractive apt-get purge -y -o Dpkg::Options::='--force-confdef' -o Dpkg::Options::='--force-confold' \
    fonts-dejavu-core \
    fonts-dejavu-mono \
  # Clean everything up to make size smaller.
  && apt-get clean \
  && apt autoclean \
  && apt autoremove \
  && rm -rf /var/lib/apt/lists/* \
  && rm -rf /usr/share/doc/*
```

This is a quick and dirty prototype, but a good proof-of-concept nevertheless.

Please learn how to master your tools before selling anything based upon biased comparisons.

**Reply by pounard** [reply]
I have to admit nonetheless that 66M is an impressive number.
